Sources/ScipioKit/ToolchainEnvironment.swift

```diff
@@ -0,0 +1,19 @@
+import struct Basics.AbsolutePath
+@_spi(SwiftPMInternal) import struct Basics.Environment
+
+public typealias ToolchainEnvironment = [String: String]
```

Comment:
@giginet I've extracted all the ToolchainEnvironment-related logic into a separate file and introduced a type alias for improved semantics. Does that work for you?